### DISCUSSION TITLE
Default fail target?

### DISCUSSION THREAD

**Discussion by MattApril** [unpoly/unpoly#403] [Q&A] [answered]
How can I specify the default `[up-fail-target]` when submitting a form? The default target seems to be `<form>`, which is picking up the first form on the page, which is not the form I want to update. In fact, I don't want to update the form at all; I want to replace the entire `<main>` or `[up-main]`.

I've tried configuring
`up.fragment.config.targets = ['[up-main]', 'main', 'body'];`

But that does not seem to work for failed requests (4xx, 5xx), and I cannot seem to find any other relevant configuration options.

EDIT: I am using v2.5.3

**Comment by adam12** [comment, marked as answer]
There is no default fail target setting that I am aware of, but you could use an [`up.macro`](https://unpoly.com/up.macro) here to set this value for you by default, on every `<form>`.

```javascript
up.macro('form', (el) => {
  el.setAttribute('up-fail-target', '[up-main]')
})
```

(ps. I edited your opening question to wrap your HTML tags in backticks)

**Reply by MattApril** [reply]
Ah ok, I meant to create some macros actually, but never got around to it. I guess now would be a good time to start. Thanks.
